tools/src/main/scala/scala/scalanative/codegen/CodeGen.scala

```diff
@@ -114,12 +118,11 @@ object CodeGen {
       (
         config.mode,
         config.LTO,
-        config.compilerConfig.incrementalCompilation
+        config.compilerConfig.useIncrementalCompilation
       ) match {
-        case (_, _, true)                                   => separateInc()
-        case (Debug, _, _)                                  => separate()
         case (ReleaseFast | ReleaseFull, build.LTO.None, _) => single()
-        case (ReleaseFast | ReleaseFull, _, _)              => separate()
+        case (_, _, true) => seperateIncrementally()
+        case _            => separate()
```

Comment:
We have only a single check for flag here. We can simplify the whole pattern matching expression by excluding boolean and checking it in the case block
```suggestion
        case (ReleaseFast | ReleaseFull, build.LTO.None) => single()
        case _ => 
           if (config.compilerConfig.useIncrementalCompilation) seperateIncrementally()
           else separate()
```